lib/cartopy/_proj4.py

```diff
@@ -0,0 +1,154 @@
+                 'rf': 'inverse_flattening',
+                 'towgs84': 'towgs84',
+                 'nadgrids': 'nadgrids'}
+PROJ_TO_CRS = {}
+
+
+def get_proj4_dict(proj4_terms):
+    """Convert a PROJ.4 string to a dictionary.
+
+    Parameters
+    ----------
+    proj4_terms: (str, dict, or iterable of key-value pairs)
+
+    Returns
+    -------
+    get_proj4_dict
+        All PROJ.4 parameters in a dictionary. Any keys with no value
+        are set to `None`.
+
+    """
+    if isinstance(proj4_terms, dict):
+        return proj4_terms
+    elif isinstance(proj4_terms, str):
+        terms = []
+        for term in proj4_terms.split(' '):
+            parts = term.strip('+').split('=')
+            if len(parts) == 1:
+                terms.append((parts[0], None))
+            else:
+                terms.append(tuple(parts[:2]))
+    else:
+        # assume list of key value pairs
+        terms = proj4_terms
+
+    return dict(terms)
+
+
+def _split_globe_parameters(proj4_dict):
+    projection_terms = {}
+    globe_terms = {}
+    for name, value in proj4_dict.items():
+        if name in _GLOBE_PARAMS:
+            globe_terms[name] = value
+        else:
+            projection_terms[name] = value
+    return projection_terms, globe_terms
+
+
+def _globe_from_proj4(globe_terms):
+    """Create a `Globe` object from PROJ.4 parameters."""
+    globe = ccrs.Globe(**{_GLOBE_PARAMS[name]: value for name, value in
+                          globe_terms.items()})
+    return globe
+
+
+class _PROJ4Projection(ccrs.Projection):
+    def __init__(self, proj4_terms, globe=None, bounds=None):
+        terms = get_proj4_dict(proj4_terms)
+        projection_terms, globe_terms = _split_globe_parameters(terms)
+        if globe is None:
+            globe = _globe_from_proj4(globe_terms)
+
+        super(_PROJ4Projection, self).__init__(projection_terms, globe)
+
+        # FIXME: Can we guess at the bounds if not provided? Maybe transform
+        #        an array of points and take the min/max of the result?
+        #        It looks like that's what LambertConformal does.
+        self.bounds = bounds
+
+    def __repr__(self):
+        return '_PROJ4Projection({})'.format(self.proj4_init)
+
+    @property
+    def boundary(self):
+        x0, x1, y0, y1 = self.bounds
+        return sgeom.LineString([(x0, y0), (x0, y1), (x1, y1), (x1, y0),
+                                 (x0, y0)])
+
+    @property
+    def x_limits(self):
+        x0, x1, y0, y1 = self.bounds
+        return (x0, x1)
+
+    @property
+    def y_limits(self):
+        x0, x1, y0, y1 = self.bounds
+        return (y0, y1)
+
+    @property
+    def threshold(self):
+        x0, x1, y0, y1 = self.bounds
+        return min(x1 - x0, y1 - y0) / 100.
+
+
+def all_subclasses(cls):
+    return cls.__subclasses__() + [g for s in cls.__subclasses__()
+                                   for g in all_subclasses(s)]
+
+
+def from_proj4(proj4_terms, globe=None, bounds=None):
```

Comment:
Is there a situation where we need to pass globe through here? It adds redundancy that we need to check against if it remains.

Reply:
You're right. Now that the default isn't to use the `_Proj4Projection` class as a last resort both of these keywords don't need to be here.